File: ESAResturant.NEW.web/ClientApp/src/app/admin/item/Models/itemPrice.ts
import { Item } from "./Item";
import { ItemSize } from "./ItemSize";
import { ItemComponent } from "./ItemComponent";

export class ItemPrice {

  constructor() {
    this.id = 0;
    this.itemId = 0;
    this.sizeId = 0;
    this.price = 0;
    this.quantityOnCart = 1;
    this.isDefullt = false;
    this.isDelete = false;
    this.itemSizeList = [];
    this.itemComponents = [];
  }


  id: number;
  itemId?: number;
  sizeId?: number;
  price: number;
  isDefullt: boolean;
  isDelete: boolean;
  item?: Item;
  itemSize?: ItemSize;
  itemComponents: ItemComponent[];

quantityOnCart: number = 1;

  itemSizeList: ItemSize[];

}

File: ESC.Resturant.Data/Entities/ItemPrice.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ESC.Resturant.Data.Entities
{
    public class ItemPrices
    {
        [Key]
        public int Id { get; set; }
        public Nullable<int> ItemId { get; set; }
        public Nullable<int> SizeId { get; set; }
        public decimal Price { get; set; }
        public bool IsDefullt { get; set; }
        public bool IsDelete { get; set; }
        public virtual Item Items { set; get; }
        public virtual ItemSize ItemSize { set; get; }
        public virtual IEnumerable<ItemComponent> ItemComponents { get; set; }
        public virtual ICollection<BillDetails> BillDetails { get; set; }


    }
}
